.github/workflows/release.yml

```diff
@@ -1,24 +1,93 @@
 name: Release
 on:
+  workflow_dispatch:
   push:
     tags:
       - "v*"
+
 jobs:
-  windows_x64_build:
-    name: Windows x64 releases
-    runs-on: windows-latest
+  build:
+    name: Build
+    strategy:
+      fail-fast: false
+      matrix:
+        target: [ windows, linux, macos]
+        include:
+          - target: windows
+            runner: windows-latest
+            channel: 'beta'  
+            artifact: build/windows/runner/Release
+          - target: linux
+            runner: ubuntu-latest
+            channel: 'stable'  
+            artifact: build/linux/x64/release/bundle
+          - target: macos
+            runner: macos-latest
+            channel: 'stable'  
+            artifact: "build/macos/Build/Products/Release/Past Paper Master.app"
+    runs-on: ${{ matrix.runner }}
+    environment: ${{ github.ref_name }}
+    timeout-minutes: 20
     steps:
-      - uses: actions/checkout@v3
-      - uses: subosito/flutter-action@v2
-        with:
-          channel: 'beta'
-      - run: flutter config --enable-windows-desktop
-      - run: flutter build windows
-  
-      - name: Create release
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          tar -a -c -f "win-PastPaperMaster-b47.zip" "build/windows/runner/Release"
-          gh release create -t="Draft release build 47" -d -F=".github/docs/release-notes.md" "Build 47" "win-PastPaperMaster-b47.zip"
-        
\ No newline at end of file
+    - name: Checkout
+      uses: actions/checkout@v3
+      
+    - name: set version number (unix)
+      run: echo "VERSION_NUM=$(echo ${{ github.ref }} | cut -d / -f 3)" >> $GITHUB_ENV
```

Comment:
Incredible work! I'm not quite sure how this version number works - is it simply a "commit number" count?

Reply:
> for tags it is refs/tags/<tag_name> ( https://docs.github.com/en/actions/learn-github-actions/contexts#github-context )

`cut -d / -f 3` 
> The cut command is used to extract sections from each line of a file or from piped input. The -d option specifies the delimiter character that separates fields in the input. The -f option specifies which field(s) to extract. The number after -f specifies the field number(s) to extract. In this case, -f 3 specifies that the third field should be extracted. [The -d / option specifies that / is the delimiter character that separates fields in the input](https://www.howtogeek.com/775824/how-to-use-the-linux-cut-command/)

check this https://github.com/WangEdward/PastPaperMaster/releases/tag/v0.1.51

Reply:
Oh, so it's a tag name. I see. Thanks for that!